assertsprocessor/factory.go

```diff
@@ -73,14 +73,17 @@ func newProcessor(logger *zap.Logger, ctx context.Context, config component.Conf
 		pConfig.DefaultLatencyThreshold = newConfig.DefaultLatencyThreshold
 	}
 
-	spanMatcher := &spanMatcher{
+	requestBuilder := &requestContextBuilderImpl{
 		logger: logger,
 	}
-	err := spanMatcher.compileRequestContextRegexps(pConfig)
+	err := requestBuilder.compileRequestContextRegexps(pConfig)
 	if err != nil {
 		return nil, err
 	}
 
+	_spanEnrichmentProcessor := buildEnrichmentProcessor(pConfig)
```

Comment:
I think this builder method can also take the `requestBuilder` as a second argument